### PR TITLE
Transform f1_elctrc_oper_rev

### PR DESCRIPTION
**Pull request by zschira**
# PR Overview
This PR covers the transformation of the `f1_elctrc_oper_rev` table. Most of the work for the structured portion of this table was just reshaping and mapping to `electric_operating_revenues_300_duration`. There's no instant table in the XBRL data, so no merging is necessary. Row #'s 22-25 in the DBF table contain unstructured data that is siloed in the XBRL data to the `electric_operating_revenues_other_300_duration`. This unstructured data is not handled by this PR.

# PR Checklist

- [x] Merge the most recent version of the branch you are merging into (probably `dev`).
- [x] All CI checks are passing. [Run tests locally to debug failures](https://catalystcoop-pudl.readthedocs.io/en/latest/dev/testing.html#running-tests-with-tox)
- [x] Make sure you've included good docstrings.
- [x] Include unit tests for new functions and classes.
- [x] Do your own explanatory review of the PR to help the reviewer understand what's going on and identify issues preemptively.

### CODE REVIEW

File: src/pudl/transform/params/ferc1.py [hunk 1]
```diff
@@ -3448,6 +3448,119 @@
...
+                        "sales_of_electricity",
+                        "provision_for_rate_refunds",
+                        "revenues_net_of_provision_for_refunds",
+                    ]
+                }
+                | {
+                    col: f"{col}_electric_operating_revenue"
+                    for col in [
+                        "residential_sales",
+                        # Note: the small_or_commercial/large_or_industrial columns are already prefixed
```
Comment (zschira) [inline src/pudl/transform/params/ferc1.py:3492]: Not sure why, but these two columns have the prefix `electric_operating_revenue` already, which is a bit inconsistent with the rest of the columns

File: src/pudl/metadata/fields.py [hunk 2]
```diff
@@ -963,6 +967,11 @@
         "type": "boolean",
         "description": "Can the generator burn multiple fuels?",
     },
+    "mwh_sold": {
+        "type": "number",
+        "description": "Megawatt hours sold.",
+        "unit": "MWh",
+    },
```
Comment (cmgosnell) [inline src/pudl/metadata/fields.py:974]: I'd suggest using `sales_mwh` instead here which is already defined. I think this is currently only being used for EIA, but it seems analogous.

File: src/pudl/metadata/resources/ferc1.py [hunk 3]
```diff
@@ -703,6 +703,28 @@
...
     },
+    "electric_operating_revenues_ferc1": {
+        "description": "Electric operating revenues - The structed part of schedule 300.",
+        "schema": {
+            "fields": [
+                "utility_id_ferc1",
+                "report_year",
+                "record_id",
+                "revenue_type",
+                "amount",
```
Comment (cmgosnell) [inline src/pudl/metadata/resources/ferc1.py:714]: if this $'s are always revenues, could this be `revenue` instead? We really should settle on how we name all of these $ columns re: #2179 .. bc we haven't really settled on anything I do think `amount` is fine if you prefer it!

**Review by cmgosnell — CHANGES_REQUESTED** [review]
It seems like the most weird part of this was the duration rename. Its very encouraging that the main work here looks like it was adding params and the dbf/xbrl row maps!

So this table has a top chunk of electricity sold w/ revenue plus MWh sold and # of customers and then a second chunk with just revenues? I think it would be nice to put that content into the resource description. So users (and us!) knows why a chunk of the MWhs & # of customer fields are null.

Or are the ones that are being dropped in the `wide_to_tidy` all of these columns in the "Other Operating Revenue" section? In these tables that have structured and unstructured portions coming from DBF we've typically pulled all of the columns that are in the XBRL structured table even if they seem like they should be the total value of the unstructured bit of the table.

File: src/pudl/transform/params/ferc1.py (hunk 1 above)
Comment (cmgosnell) [reply src/pudl/transform/params/ferc1.py:3492]: lol. that is weird but also is consistent with FERCs inconsistency.

File: src/pudl/package_data/settings/etl_fast.yml [hunk 4]
```diff
@@ -73,8 +74,12 @@ ferc_to_sqlite_settings:
       - electric_operations_and_maintenance_expenses_320_instant
       - retained_earnings_118_duration
       - retained_earnings_118_instant
-      - statement_of_cash_flows_120_duration
+      - electric_operating_revenues_300_duration
+      - electric_operating_revenues_300_instant
+      - electric_operating_revenues_other_300_duration
+      - electric_operating_revenues_300_instant
```
[the remark this answers is not in the document]
Comment (zschira) [reply src/pudl/package_data/settings/etl_fast.yml:80]: I was using the `other` tables (and had a typo) when I was messing around with the unstructured data, but don't need them now, so I removed these

File: src/pudl/transform/params/ferc1.py [hunk 5]
```diff
@@ -3448,6 +3448,119 @@
...
+                "idx_cols": [
+                    "entity_id",
+                    "report_year",
+                ],
+                "value_types": [
+                    "electric_operating_revenue",
+                    "mwh_sold",
+                    "avg_customers_per_month",
+                ],
+                "expected_drop_cols": 14,
```
[the remark this answers is not in the document]
Comment (zschira) [reply src/pudl/transform/params/ferc1.py]: These were actually columns that didn't have the mwh sold and average customers, that I was leaving out while deciding how to handle them, and forgot to add them back in. I've added these columns back, and had to add a small bit of code to drop some records with duplicate primary keys. The records in question were all from one utility in 2011 and every one of them had a revenue of 3.33e8 or 3.333e9 and all other fields were Null. These seemed pretty meaningless, so I dropped them all.

File: src/pudl/metadata/fields.py (hunk 2 above)
Comment (zschira) [reply src/pudl/metadata/fields.py]: Changed

File: src/pudl/metadata/resources/ferc1.py (hunk 3 above)
Comment (zschira) [reply src/pudl/metadata/resources/ferc1.py]: I actually prefer `revenue`, and had considered that initially, but went with `amount` to try and be consistent with other tables. I've changed to `revenue` now.

**Comment by zschira** [comment]
> So this table has a top chunk of electricity sold w/ revenue plus MWh sold and # of customers and then a second chunk with just revenues? I think it would be nice to put that content into the resource description. So users (and us!) knows why a chunk of the MWhs & # of customer fields are null

Yeah that's correct. I've added a note in the description about the NULL fields on many records, and added notes in `dbf_to_xbrl.csv` indicating which `revenue_type`'s this applies to. Does this seem sufficient?

**Review by cmgosnell — APPROVED** [review]
this looks great! glad you slurped the revenue-only bit back in and I think your description in the resources is sufficient for sure!! approved (pending you merge dev in and the tests pass of course!)

**Pull request merged**